support extendr_polars

sorhawell opened the pull request:
This PR will add support for [`extendr_polars`](https://github.com/rpolars/extendr_polars) which allows rust extensions to polars in R. [Examples](https://github.com/rpolars/extendrpolarsexamples)

sorhawell commented:
this update is needed  to support the crate [extendr_polars](https://github.com/rpolars/extendr_polars) 

`extendr_polars` crate allows an R package developer to interface with the polars R package via  arrow-import/export r-polars of DataFrame. This is zero cost and much safer than transmuting the inner rust-polars DataFrame between compilation units and differing versions.

an example R package [is found here](https://github.com/rpolars/extendrpolarsexamples)

etiennebacher reviewed (approved):
@sorhawell I can approve if you want but I have no clue if this is good or if there are mistake. I suppose it works fine since you made some tests repos using it, but maybe @eitsupi would be more comfortable reviewing this?

eitsupi commented on src/rust/src/arrow_interop/to_rust.rs:
```diff
@@ -160,3 +134,63 @@ pub fn to_rust_df(rb: Robj) -> Result<pl::DataFrame, String> {
+}
+
+// implementation of consuming stream to Series. Stream is drop/released hereafter.
+fn consume_arrow_stream_to_s(boxed_stream: Box<ffi::ArrowArrayStream>) -> RResult<pl::Series> {
```
`to_s` may be a little ambiguous?

sorhawell replied:
thx I have renamed  to `...to_series`

eitsupi reviewed (commented):
Sorry, I don't have enough knowledge to review this.......

sorhawell commented:
@etiennebacher  arrow-protocol is really hairy ^^ I have looked in the sparse official arrow docs for days to get an ok understanding. 

I have added a round trip unit test with [b38f4d8](https://github.com/pola-rs/r-polars/pull/326/commits/b38f4d8194fdc793aa02080a2c9b6f2853c49c40) and added some notes for what is going on and when to use this.

eitsupi commented:
Thanks!
